src/darwin/Framework/CHIP/MTRDeviceControllerXPCConnection.mm

```diff
@@ -67,9 +67,45 @@ @implementation MTRDeviceControllerXPCConnection
 
 - (instancetype)initWithWorkQueue:(dispatch_queue_t)workQueue connectBlock:(NSXPCConnection * (^)(void) )connectBlock
 {
+    // Classes that are used by MTRDevice responses.  In particular, needs to
+    // include NSError.
+    //
+    // TODO: should this include NSOrderedSet, NSSet, NSDate, NSNull, even
+    // though we don't use them?  Should we have public API for some sort for
+    // this set?
+    __auto_type * allowedClasses = [NSSet setWithArray:@[
```

Comment:
I think we should aim to use specific class sets for particular use cases, so something like `allowedAttributeValueClasses` might be more descriptive. Where would sets / ordered sets etc come from? 

Allowing clients to change this set via an API sounds dubious to me, do you have a specific use case in mind?

Reply:
> so something like allowedAttributeValueClasses might be more descriptive.

It's not just attribute values; it's all possible responses from the MTRDevice stuff, including command responses, etc.

> Where would sets / ordered sets etc come from?

They wouldn't, unless we changed something about the API....

> Allowing clients to change this set via an API sounds dubious to me, do you have a specific use case in mind?

I meant an API for _getting_ this set, so that API consumers who are trying to do this XPC stuff don't have to re-derive this for themselves.  I guess really we'd want an API for configuring the relevant NSXPCInterface or something.

Reply:
I went with an API for getting and configuring the `NSXPCInterface` instances, since that's what consumers need in practice.